include/chemfiles/files/DCDFile.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef CHEMFILES_DCD_FILE_HPP
+#define CHEMFILES_DCD_FILE_HPP
+
+#include "chemfiles/File.hpp"
+#include <cstdint>
+#include <fstream>
+#include <string>
+ 
+namespace chemfiles {
+
+   /// Handle the creation and destruction of the file as needed.
+   /// Reads the file header and stores the offsets for individual frames.
+   class DCDFile {
+   public:
+
+      DCDFile(std::string path, File::Mode mode);
+      ~DCDFile();
+
+      void                          readHeader();
+      void                          readFrame(std::vector<float>& X, std::vector<float>& Y,
+                                                std::vector<float>& Z, int toFrame = NULL);
+      void                          writeHeader();
+      void                          writeFrame(std::vector<float>& x, std::vector<float>& y, 
+                                                std::vector<float>& z);
+
+      void                          setHDR(std::string hdr);
+      void                          setProperties(std::vector<int>& properties);
+
+      int                           getNumFrames() const;
+      
+      //Information on the periodic boundary conditions of the system
+      bool                          hasCrystal() const;
+      const double*                 getPbc() const;
+      void                          setPbc(double* pbc);
+
+      //
+      bool                          hasFrozenAtoms();
+      int                           getNumFreeAtoms() const;
+      void                          setNumFreeAtoms(int freeAtomsNum);
+      void                          setFreeAtoms(std::vector<int>& freeAtomsIndexes);
+      
+      //Number of atoms in the system 
+      int                           getNumAtoms() const;
+      void                          setNumAtoms(size_t atomNum);
+
+      bool                          hasPositions() const;
+      
+      //Navigate in the file stream
+      virtual size_t                getPos();
+      void                          setPos(size_t frameNum);
+
+      //Read title information 
+      int                           getNumTitles();
+      void                          setNumTitles(int titlesNumber);
+      char*                         getTitle();
+      void                          setTitle(std::string title);
+      
+   protected:
```

Comment:
Private would be okay since DCDFile class doesn't have any derived/sub/child class. It was more by "habit" that I have chosen protected instead of private.